pkg/judge/judge.go

```diff
@@ -0,0 +1,128 @@
+package judge
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/jasondellaluce/synchro/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+type ChangeType int
+
+const (
+	Modify ChangeType = iota
+	Add
+	Delete
+	Rename
+)
+
+type CommitChange struct {
+	FileName string
+	Type     ChangeType
+}
+
+func Judge(ctx context.Context, git utils.GitHelper, commit string) error {
+	out, err := git.DoOutput("show", "--raw", commit)
+	if err != nil {
+		return err
+	}
+
+	modified := 0
+	added := 0
+	deleted := 0
+	renamed := 0
+
+	lines := strings.Split(out, "\n")
+	var changes []CommitChange
+
+	for _, line := range lines {
+		if len(line) > 0 && line[0] == ':' {
+			change, err := parseMetadataInfo(line)
+			if err != nil {
+				return err
+			}
+			changes = append(changes, change)
+		}
+	}
+
+	for _, change := range changes {
+		switch change.Type {
+		case Modify:
+			modified = 1
+		case Add:
+			added = 1
+		case Delete:
+			deleted = 1
+		case Rename:
+			renamed = 1
+		}
+	}
+
+	if (modified + added + deleted + renamed) > 1 {
+		logrus.Errorf("commits can either have modified, added, deleted or renamed files and not a combination of those")
```

Comment:
```suggestion
		err = errors.New("commits must have either modified, added, deleted or renamed files, and not a combination of those")
		fmt.Fprintln(os.Stdout, err.Error())
```
Logrus would print to stderr. This way we print to stdout, plus we return the message also as an error